images/orbit-controller/src/orbit_controller/operators/orbitjob_operator.py

```diff
@@ -167,3 +167,54 @@ def create_job(
         "orbitJobOperator": {"jobStatus": "JobCreated", "jobName": job_instance_metadata.name, "nodeType": node_type}
     }
     return "JobCreated"
+
+
+@kopf.on.timer(
+    ORBIT_API_GROUP,
+    ORBIT_API_VERSION,
+    "orbitjobs",
+    interval=5,
+    initial_delay=20
+    # when=_should_index_jobs
+)
+def orbit_job_monitor(
+    namespace: str,
+    name: str,
+    patch: kopf.Patch,
+    logger: kopf.Logger,
+    namespaces_idx: kopf.Index[str, Dict[str, Any]],
+    jobs_idx: kopf.Index[Tuple[str, str], Dict[str, Any]],
+    **_: Any,
+) -> Any:
+    ns: Optional[Dict[str, Any]] = None
+    k8s_job: Optional[Dict[str, Any]] = None
+
+    for ns in namespaces_idx.get(namespace, []):
+        logger.debug("ns: %s", ns)
+
+    if ns is None:
+        patch["status"] = {
+            "orbitJobOperator": {"jobStatus": "JobDetailsNotFound", "error": "No Namespace resource found"}
+        }
+        return "JobDetailsNotFound"
+
+    for k8s_job in jobs_idx.get((namespace, name), []):
+        logger.debug("k8s_job: %s", k8s_job)
+
+    if k8s_job is None:
+        patch["status"] = {"orbitJobOperator": {"jobStatus": "JobMetadataNotFound", "error": "No Job Metadata found"}}
+        return "JobMetadataNotFound"
+
+    jobStatus = k8s_job.get("status").get("conditions")[0].get("type")  # type: ignore
```

Comment:
this will throw an exception if the status does not contain a `conditions` attribute with a value of an array with at least one element. we know that there are times (e.g. initial creation) when this will not be the case. this should contain some error handling like:
```
jobStatus = k8s_job.get("status", {}).get("conditions", [{}])[0].get("type")
```
where we return a default value with the `.get()` call. without a default it will return `None`

also, `jobStatus` should be `job_status` to conform to our linting checks

this all applies to the following 2 lines as well

Reply:
Thanks for catching this one. Fixing it